code/modules/urist/mob/uristliving.dm

```diff
@@ -0,0 +1,64 @@
+//vg color-matrix code
+/mob/proc/get_screen_colour()
+	if(!client)
+		return 0
+	if(M_NOIR in mutations)
+		return NOIRMATRIX
+
+/mob/dead/observer/get_screen_colour()
+	return default_colour_matrix
+
+/mob/living/simple_animal/get_screen_colour()
+	. = ..()
+	if(.)
+		return .
+	else if(src.colourmatrix.len)
+		return src.colourmatrix
+
+/mob/living/carbon/human/get_screen_colour()
+	. = ..()
+	if(.)
+		return .
+	//var/datum/organ/internal/eyes/eyes = internal_organs_by_name["eyes"]
+	//if(eyes && eyes.colourmatrix.len && !(eyes.robotic))
+	//	return eyes.colourmatrix
+	//else return
+	return default_colour_matrix
+
+/mob/proc/update_colour(var/time = 50,var/forceupdate = 0)
+	if(!client || (client.updating_colour && !forceupdate))
+		return
+	var/list/colour_to_apply = get_screen_colour()
+	var/list/difference = difflist(client.color,colour_to_apply)
+	if(difference || !(client.color) || !istype(difference) || !difference.len)
+		client.updating_colour = 1
+		var/cached_ckey = client.ckey
+		if(forceupdate)
+			time = 0
+		else if(colour_to_apply == NOIRMATRIX)
+			time = 170
+		client.colour_transition(colour_to_apply,time = time)
+		spawn(time)
+			if(client && client.mob != src)
+				return
+			if(client)
+				client.color = colour_to_apply
+				client.updating_colour = 0
+				difference = difflist(client.color,get_screen_colour())
+				if((difference || !(client.color) || !istype(difference) || !difference.len) && !forceupdate) // panic panic panic
+					src.update_colour(forceupdate = 1)
+			else
+				bad_changing_colour_ckeys["[cached_ckey]"] = 1
+
+/mob/proc/handle_urist_hooks()
+	return
+
+/mob/living/handle_urist_hooks()
+	. = ..()
+	update_colour()
+	return .
+
+/mob/living/Life()
```

Comment:
Good question, I'll have to go through the logic - this is something like a three year old bit of /vg/ code. 
I jumped the gun a bit on notifying the two of you, since I just threw a rebase at it, cleaned up the most egregious problems and factored it out of upstream modules; I'm working on polishing up the actual logic now.

Reply:
Still fairly sure this update isn't needed every life tick. If for some reason it is then you'll need to move it over to [https://github.com/UristMcStation/UristMcStation/blob/master/code/modules/mob/living/life.dm](url).

Reply:
I don't believe I need to move it; DM supports this weird flavor of modular extension, and putting it into core files is very non-kosher for merges.

It's not strictly needed, but it's returning very early if there's nothing to change anyway. What it should be doing is to have an event listener on client changes, but both the docs and the commit history on observers have failed me when it comes to providing a clear interface spec. And again: Three. Years. Unless it has a huge performance impact, I can just update it in a later PR.

Reply:
Good enough reasoning for me.